File: server/src/main/java/au/com/codeka/warworlds/server/admin/Session.java
package au.com.codeka.warworlds.server.admin;

import org.joda.time.DateTime;

import javax.annotation.Nonnull;

import au.com.codeka.warworlds.common.proto.AdminRole;
import au.com.codeka.warworlds.common.proto.AdminUser;

/**
 * Represents details about the current session, such as your username, access level and so on.
 */
public class Session {
  private final String cookie;
  private final DateTime loginTime;
  @Nonnull private final AdminUser adminUser;

  public Session(String cookie, @Nonnull AdminUser adminUser, DateTime loginTime) {
    this.cookie = cookie;
    this.adminUser = adminUser;
    this.loginTime = loginTime;
  }

  public String getCookie() {
    return cookie;
  }

  public String getEmail() {
    return adminUser.email_addr;
  }

  public boolean isInRole(AdminRole role) {
    for (AdminRole ar : adminUser.roles) {
      if (ar.equals(role)) {
        return true;
      }
    }
    return false;
  }

  public DateTime getLoginTime() {
    return loginTime;
  }
}

File: server/src/main/kotlin/au/com/codeka/warworlds/server/admin/Session.kt
package au.com.codeka.warworlds.server.admin

import au.com.codeka.warworlds.common.proto.AdminRole
import au.com.codeka.warworlds.common.proto.AdminUser
import org.joda.time.DateTime

/**
 * Represents details about the current session, such as your username, access level and so on.
 */
class Session(val cookie: String, private val adminUser: AdminUser?, val loginTime: DateTime) {

  val email: String
    get() = adminUser!!.email_addr

  fun isInRole(role: AdminRole): Boolean {
    for (ar in adminUser!!.roles) {
      if (ar == role) {
        return true
      }
    }
    return false
  }

}